templates/base_index.html

```diff
@@ -1409,56 +1409,83 @@ <h2>
     * Choose a takeover
     * ===
     *
-    * From the list of provided takeovers in the #takeovers template,
+    * From the list of provided takeovers from /.takeovers.json,
     * choose one (that matches the client's language), and replace the
     * base template with it.
     */
 
-    var baseTakeover = document.getElementById('takeover');  // The base takeover element
-    var takeovers = document.getElementById('takeovers');  // The list of current takeovers to choose from
-
-    // For browsers that support <template> (all but IE), grab the template's ".content"
-    takeovers = 'content' in takeovers ? takeovers.content: takeovers;
+    fetch("/takeovers.json").then(response => response.json()).then(takeovers => {
 
-    // First, select takeovers that don't specify a language and don't exclude the users language
-    var takeoverSelectors = [".js-takeover:not([lang]).js-takeover:not([lang-skip*=" + primaryParentLanguage + "])"]
 
-    // Add selectors to find takeovers for any of the user's languages
-    takeoverSelectors.push(".js-takeover[lang=" + primaryParentLanguage + "]");
-    takeoverSelectors.push(".js-takeover[data-lang-extra*=" + primaryParentLanguage + "]");
+    var baseTakeover = document.getElementById('takeover');  // The base takeover element
+    
 
     // Get the selected takeovers
-    var selectedTakeovers = (takeovers.querySelectorAll(takeoverSelectors.join(',')));
+    var selectedTakeovers = takeovers.filter(function(item) 
+       {
+        return item.lang === primaryParentLanguage || item.lang === ""
+      })
 
-    if (selectedTakeovers) {
+      if (selectedTakeovers && selectedTakeovers.length > 0) {
       var selectedIndex = null;
 
       if (localStorage.getItem('selected_takeover_index') !== null) {
         // If we previously chose a takeover, increment the number to show the next takeover
         var nextIndex = parseInt(localStorage.getItem('selected_takeover_index')) + 1
         selectedIndex = nextIndex < selectedTakeovers.length ? nextIndex : 0;
+        showTakeover(selectedTakeovers, selectedIndex);
       } else {
         // Otherwise, randomly choose one of the takeovers and store it for next time
         selectedIndex = Math.floor(Math.random() * selectedTakeovers.length);
+        showTakeover(selectedTakeovers, selectedIndex);
       }
 
       // Store the current takeover
       localStorage.setItem('selected_takeover_index', selectedIndex)
 
-      // Get the takeover element
-      var selectedTakeover = selectedTakeovers[selectedIndex]
+    }
+    });
+  }
 
-      if (! document.body.contains(selectedTakeover)) {
-        // If it's not in the current document (as in, it's in a template) import it
-        // Note: This is for all browser *except* IE, which doesn't support <template>s
-        selectedTakeover = document.importNode(selectedTakeover, true);
-      }
+  function showTakeover(takeovers, index = 0) {
```

Comment:
That's quite a lot of JS to keep inline on the page. Ideally it should be in a separate file.

And it seems it could use some transpiling as well. There are a lot of ES6 features that will fail in older browsers.

Most notably:
- default values of function parameters
- template literals ` `${...}` `

Other things that may fail in older IE
- const
- .classList API

If we want to keep it as inline script it should at least be written in most cross-browser way, avoiding unnecessary ES6 features.

Reply:
Rewritten with old JS. Should support IE > 9